plugins/modules/keycloak_user_federation.py

```diff
@@ -93,6 +93,16 @@
         default: true
         version_added: 9.4.0
 
+    exclude_bind_credential_from_update_check:
+        description:
+            - The value of the config parameter O(config.bindCredential) is redacted in the Keycloak responses. Comparing
+              the redacted value with the desired value always evaluates to not equal.
+            - Set to V(true) to exclude O(config.bindCredential) when comparing the before state with the desired state to
+              determine wether an update is required.
+        type: bool
+        default: false
```

Comment:
Hmm, how about using a string state with two states (choices) instead of a boolean? Then it is possible to later add more options, if for example the API is changed to allow other ways to handle this.

Reply:
Yeah, sounds good. I'm not sure about the naming though.

Maybe
```
bind_credential_handling:
    description:
        - The value of the config parameter O(config.bindCredential) is redacted in the Keycloak responses. There is currently no way to get the actual value. Comparing the redacted value with the desired value always evaluates to not equal. This also means the before and desired states are never equal if the parameter is set.
        - Set to V(exclude_from_update_check) to exclude O(config.bindCredential) when comparing the before state with the desired state to determine whether an update is required.
    type: str
    default: include_in_update_check
    choices:
        - include_in_update_check
        - exclude_from_update_check
        (- get_cleartext_value)
```

Reply:
I would probably adjust the option name a bit - maybe `bind_credential_update_mode`? - and simplify the choices: `never` (instead of `exclude`), `always` (instead of `update`). WDYT?

Reply:
I like the option name, but i'm not sure about `never`. So far the PR only removes the parameter from the update check not the actual update. If there are other changes the parameter would still get updated, so never seems kind of misleading i think. We could adjust the PR accordingly though. I think this would also make a bindCredential update more predictable, either do an update or not. Comparing the redacted value is kind of meaningless. If there is a way to get the actual value later we could then add `on_change`.

Reply:
True, `never` isn't the best name. How about `only_indirect`? I think it's a bit easier to understand than `skip_check`. (I could also be wrong/biased though :) )

Reply:
You're right, I think we got a winner. I will adjust the PR.